packages/serverpod_shared/lib/src/config.dart

```diff
@@ -436,7 +436,10 @@ class DatabaseConfig {
 
     var password = passwords[ServerpodPassword.databasePassword.configKey];
     if (password == null) {
-      throw Exception('Missing database password.');
+      throw DatabasePasswordMissingException(
+        ServerpodPassword.databasePassword.configKey,
+        'SERVERPOD_PASSWORD_database',
```

Comment:
The `ENV` key can always be derived from the YAML config one, right?

Then maybe we should not take an extra parameter for now.

Reply:
For ones that already had an env key (before the move to `SERVERPOD_PASSWORD_*`), we still have the old keys for backward compatibility. So for database, it's `SERVERPOD_DATABASE_PASSWORD`. Ideally, we want to give as much information to the user about what went wrong, no? Which is why I added the `envVarName` field to print this information